setup.py

```diff
@@ -1,59 +1,10 @@
 """
```

Comment:
setup.py is no longer required since setuptools 64.0.0 (https://setuptools.pypa.io/en/latest/userguide/development_mode.html)

Reply:
Cool, it seems like setuptools v64.0.0 enabled non setup.py editable installs!

On the developer side, this seems ok to remove setup.py, but are there any potential side effects for users not having such a new setuptools version? Or to put it another way, do we want to remove setup.py for PyGMT v0.8.0, or wait another minor version (e.g. v0.9.0)?

Reply:
> are there any potential side effects for users not having such a new setuptools version?

I have setuptools 62.6.0 installed. Checking out this branch and running `make install` and `make package` work well for me.

When running `make package`, I see the following messages:
```
$ make package 
python -m build --sdist --wheel
* Creating venv isolated environment...
* Installing packages in isolated environment... (setuptools>=64, setuptools_scm[toml]>=6.2)
* Getting dependencies for sdist...
/tmp/build-env-ld441dbo/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py:108: _BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
  warnings.warn(msg, _BetaConfiguration)
running egg_info
writing pygmt.egg-info/PKG-INFO
writing dependency_links to pygmt.egg-info/dependency_links.txt
writing requirements to pygmt.egg-info/requires.txt
writing top-level names to pygmt.egg-info/top_level.txt
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE.txt'
adding license file 'AUTHORS.md'
adding license file 'AUTHORSHIP.md'
writing manifest file 'pygmt.egg-info/SOURCES.txt'
* Building sdist...
/tmp/build-env-ld441dbo/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py:108: _BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
  warnings.warn(msg, _BetaConfiguration)
running sdist
running egg_info
writing pygmt.egg-info/PKG-INFO
writing dependency_links to pygmt.egg-info/dependency_links.txt
writing requirements to pygmt.egg-info/requires.txt
writing top-level names to pygmt.egg-info/top_level.txt
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE.txt'
adding license file 'AUTHORS.md'
adding license file 'AUTHORSHIP.md'
writing manifest file 'pygmt.egg-info/SOURCES.txt'
running check
...
```
It seems build dependencies like setuptools>=64 are automatically installed in isolated environments and do not affect users' installations.

Reply:
A downside here is that `make install` no longer works without internet access, even if setuptools>=64 is installed in the environment. I'll do some more searching for a workaround but if anyone knows a solution that would be helpful.

Reply:
I tried disconnecting and used `pip install --no-deps --no-build-isolation --editable=.` which seems to work offline (presumably it manages to pick up a pre-downloaded `setuptools>=64`). However, it results in a editable version named `pygmt-0.0.0`.

Reply:
> A downside here is that `make install` no longer works without internet access, even if setuptools>=64 is installed in the environment.

It's unrelated to this PR. Offline `make install` worked for pygmt<=0.2.0 and started to fail since v0.3.0, in which we switched from versioneer to setuptools_scm (https://www.pygmt.org/dev/changes.html#id47). So it's likely a problem for setuptools_scm.

> I tried disconnecting and used `pip install --no-deps --no-build-isolation --editable=.` which seems to work offline (presumably it manages to pick up a pre-downloaded `setuptools>=64`). However, it results in a editable version named `pygmt-0.0.0`.

This command gives the correct version for pygmt <= 0.6.0 but results in `pygmt-0.0.0` for pygmt v0.7.0. Not sure if it's caused by changes in https://github.com/GenericMappingTools/pygmt/pull/1945.

Reply:
@maxrjones Perhaps open an issue report for the offline `pip` issue?

Reply:
I've opened a PR in https://github.com/GenericMappingTools/pygmt/issues/2097.